gateway/session_manager.go

```diff
@@ -234,41 +237,61 @@ func (l *SessionLimiter) ForwardMessage(r *http.Request, session *user.SessionSt
 		return sessionFailRateLimit
 	}
 
+	var (
+		apiLimit            = accessDef.Limit.Clone()
+		endpointRLKeySuffix = ""
+	)
+
+	if len(accessDef.Endpoints) > 0 {
+		endpointRLInfo, doEndpointRL := getEndpointRateLimitInfo(r.Method, r.URL.Path, accessDef.Endpoints)
+		if doEndpointRL {
+			apiLimit.Rate = float64(endpointRLInfo.rate)
```

Comment:
if you reuse the type you can avoid this cast; could use a type alias if you still want to keep a local type symbol